src/pages/messages/components/NviCandidateActions.tsx

```diff
@@ -233,30 +243,94 @@ export const NviCandidateActions = ({ nviCandidate, nviCandidateQueryKey }: NviC
+              onAccept={async (reason) => {
+                await statusMutation.mutateAsync({ status: 'Rejected', reason });
+                setHasSelectedRejectCandidate(false);
+              }}
+              isLoading={statusMutation.isPending}
+            />
 
-            {hasSelectedRejectCandidate && (
-              <MessageForm
-                confirmAction={async (reason) => {
-                  await statusMutation.mutateAsync({ status: 'Rejected', reason });
-                  setHasSelectedRejectCandidate(false);
-                }}
-                cancelAction={() => setHasSelectedRejectCandidate(false)}
-                fieldLabel={t('tasks.nvi.reject_nvi_candidate_form_label')}
-                buttonTitle={t('tasks.nvi.reject_nvi_candidate')}
-              />
-            )}
+            <Divider sx={{ mb: '1rem' }} />
           </>
         )}
 
-        {!hasSelectedRejectCandidate && (
-          <MessageForm
-            confirmAction={async (text) => await createNoteMutation.mutateAsync({ text })}
-            fieldLabel={t('tasks.nvi.note')}
-            buttonTitle={t('tasks.nvi.save_note')}
-          />
-        )}
+        <MessageForm
+          confirmAction={async (text) => await createNoteMutation.mutateAsync({ text })}
+          fieldLabel={t('tasks.nvi.note')}
+          buttonTitle={t('tasks.nvi.save_note')}
+        />
       </Box>
     </>
   );
 };
+
+interface RejectionDialogProps {
+  open: boolean;
+  onCancel: () => void;
+  onAccept: (reason: string) => Promise<unknown> | void;
+  isLoading: boolean;
+}
+
+const maxLength = 160;
+
+const RejectionDialog = ({ open, onCancel, onAccept, isLoading }: RejectionDialogProps) => {
+  const { t } = useTranslation();
+  const [reason, setReason] = useState('');
+
+  const handleAccept = async () => {
+    await onAccept(reason.trim());
+    setReason('');
+  };
+
+  const handleClose = () => {
+    onCancel();
+    setReason('');
+  };
+
+  const handleChange = (event: ChangeEvent<HTMLInputElement>) => {
+    setReason(event.target.value);
+  };
+
+  return (
+    <Dialog open={open} onClose={handleClose}>
+      <DialogTitle>{t('tasks.nvi.reject_nvi_candidate')}</DialogTitle>
+      <DialogContent>
+        <Typography gutterBottom>{t('tasks.nvi.reject_nvi_candidate_modal_text')}</Typography>
+        <form onSubmit={handleAccept}>
+          <TextField
+            value={reason}
+            onChange={handleChange}
+            data-testid={dataTestId.tasksPage.nvi.rejectionModalTextField}
+            inputProps={{ maxLength: maxLength }}
+            variant="filled"
+            multiline
+            minRows={3}
+            maxRows={Infinity}
+            fullWidth
+            required
+            label={t('tasks.nvi.reject_nvi_candidate_form_label')}
+            helperText={`${reason.length}/${maxLength}`}
+            FormHelperTextProps={{ sx: { textAlign: 'end' } }}
+          />
+
+          <DialogActions>
```

Comment:
Har knota alt for mye DialogContent og DialogActions selv, og noen ganger har jeg bare ansett slaget som tapt og gått for noe som virker 😩 
Men ser ut som det skal være greit å få det riktig her, uten å wrappe de inni hverandre.

```
<form>
  <DialogContent> ... </DialogContent>
  <DialogActions> ... </DialogActions>
</form>
```

Dette for å unngå en dobbel padding på det som ligger i Actions og litt sånt